packages/financial-templates-lib/src/clients/FinancialContractClient.ts

```diff
@@ -22,81 +87,52 @@ class FinancialContractClient {
    * @return None or throws an Error.
    */
   constructor(
-    logger,
-    financialContractAbi,
-    web3,
-    financialContractAddress,
-    multicallContractAddress,
+    private readonly logger: Logger,
+    financialContractAbi: Abi,
+    readonly web3: Web3,
+    private readonly financialContractAddress: string,
+    // We use this Multicall contract to simulate the results of batched state-modifying transactions.
+    private readonly multicallContractAddress: string,
     collateralDecimals = 18,
     syntheticDecimals = 18,
     priceFeedDecimals = 18,
-    contractType = "ExpiringMultiParty"
-    // Default to Expiring Multi Party for now to enable backwards compatibility with other bots.
-    // This will be removed as soon as the other bots have been updated to work with these contract types.
+    private readonly contractType: FinancialContractType = "ExpiringMultiParty" // Default to Expiring Multi Party for now to enable backwards compatibility with other bots. // This will be removed as soon as the other bots have been updated to work with these contract types.
   ) {
-    this.logger = logger;
-    this.web3 = web3;
-
     // Financial Contract contract
-    this.financialContract = new web3.eth.Contract(financialContractAbi, financialContractAddress);
-    this.financialContractAddress = financialContractAddress;
-    this.financialContractAbi = financialContractAbi;
-
-    // We use this Multicall contract to simulate the results of batched state-modifying transactions.
-    this.multicallContractAddress = multicallContractAddress;
-
-    // Financial Contract Data structures & values to enable synchronous returns of the financialContract state seen by
-    // the client.
-    this.activeSponsors = [];
-    this.positions = [];
-    this.undisputedLiquidations = [];
-    this.expiredLiquidations = [];
-    this.disputedLiquidations = [];
-    this.collateralRequirement = null;
-    this.collateralDecimals = collateralDecimals;
-    this.syntheticDecimals = syntheticDecimals;
-    this.priceFeedDecimals = priceFeedDecimals;
-
-    // Perpetual financial products require the latest fundingRate to correctly calculate fundingRate adjusted debt.
-    this.latestCumulativeFundingRateMultiplier;
-
-    // Store the last on-chain time the clients were updated to inform price request information.
-    this.lastUpdateTimestamp = 0;
-
-    // Helper functions from web3.
-    this.toBN = this.web3.utils.toBN;
-    this.toWei = this.web3.utils.toWei;
+    this.financialContract = (new web3.eth.Contract(
+      financialContractAbi,
+      financialContractAddress
+    ) as unknown) as FinancialContractClient["financialContract"]; // Cast to web3-specific type
```

Comment:
why can this not just be `as  ExpiringMultiPartyWeb3 | PerpetualWeb3`? There is similar syntax where you do look up with the `["..."]`. is there a reason you do this indirectly rather than just using the type directly?

Reply:
No, you're right. I was just inconsistent. I'll fix this.

Reply:
Fixed.